glean-core/ffi/src/lib.rs

```diff
@@ -198,16 +196,51 @@ pub extern "C" fn glean_enable_logging() {
     }
 }
 
-#[no_mangle]
-pub extern "C" fn glean_initialize(
-    data_dir: FfiStr,
-    application_id: FfiStr,
+#[repr(C)]
+pub struct FfiConfiguration<'a> {
```

Comment:
nit: I'd drop a comment here stating that if this is changed, the platform specific code needs to be changed as well (similarly to the comment on the Kotlin side)